core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1028,6 +1028,7 @@ class ReplicaManager(val config: KafkaConfig,
     var bytesReadable: Long = 0
     var errorReadingData = false
     var hasDivergingEpoch = false
+    var notLocalPreferredReadReplica = false
```

Comment:
There is no impact at present. I need to use the replica read capability to do some development, so I added a little log to readFromLog() to verify the correctness. I found that every time the leader elects a non-leader replica, it will trigger the fetch response and return with a delay of 500 ms , even though I'm consuming historical data. Triggered my log to print, and then traced this case

Reply:
There is no impact at present. 
I need to use the replica read capability to do some development, so I added a little log to readFromLog() to verify the correctness. I found that every time the leader broker selects a non-leader replica, it will trigger the fetch response  return with a delay of 500 ms , even though I'm consuming historical data. Triggered my log to print, and then traced this case
I make some code modifications and add unit tests according to your suggestion.

Reply:
Interesting.. I thought that the delayed fetch would return immediately in this case. Looking at the code, it does not because the fetch offset is set to `UnknownOffsetMetadata` so the delayed fetch just ignore it. I think that your patch makes total sense. Let me know when the comments are addressed.

Reply:
@dajac added a unit test and made some code modifications.